src/bin/gosub-parser.rs

```diff
@@ -42,8 +42,7 @@ fn main() -> Result<()> {
     }
 
     let document = DocumentBuilder::new_document();
-    let mut parser = Html5Parser::new(&mut stream, document.clone());
-    let parse_errors = parser.parse()?;
+    let parse_errors = Html5Parser::parse_document(&mut stream, Document::clone(&document), None)?;
```

Comment:
We don't call a "new()" method first to initialize a parser, but directly call the `parse_document()` or `parse_document_fragment()` instead.  This is due the fact that we need to initialize the parser, but we cannot do this until we actually call the parser (otherwise we have to specify things with `option<>`, which doesn't sound right either.

Reply:
I'd ultimately switch to something that looks a little like this:
```rust
// Parse a whole document
let result = gosub::html5::parse(&mut stream, Document::clone(&document), None)?;

// Parse a fragment
let result = gosub::html5::parse_fragment(&mut stream, Document::clone(&document) /*, ... */)?;
```

Since we already have the `html5` module, `Html5Parser` adds stutter, and `html::Parser` would be simpler.  But in this case, we can just expose two functions, `html5::parse` and `html::parse_fragment`, that delegate to a `Parser` without making it part of the public interface except in more advanced use cases.